### ISSUE TITLE
libnautilus: Unknown directive found `pragma`

### ISSUE THREAD

**Issue by matheusfaustino** [ircmaxell/php-c-parser#5]
Hi, first of all, thank you for your work and your article, I was about to give up about FFI and then I found your article and libs and I'm giving another chance for it.

I am trying to load the nautilus extension with FFIMe, but it throws this error:
`Fatal error: Uncaught LogicException: Unknown directive found pragma in ./vendor/ircmaxell/php-c-parser/lib/PreProcessor.php:114`

The code I'm running is:
```
$ffime = (new FFIMe('/usr/lib/x86_64-linux-gnu/libnautilus-extension.so'))
    ->include('/usr/include/nautilus/nautilus-extension.h')
    ->codeGen('nautilus\\extension', __DIR__ . '/nautilius.php');
```

if I remove the nautilus-extension.h it compiles, but it doesn't generate all the functions. You can find more about  [here nautilus-extension.h](https://github.com/GNOME/nautilus/blob/master/libnautilus-extension/nautilus-extension.h) . The actual header doesn't have the preprocessor `pragma` but the others do. 

Could you help me, please?

**Comment by matheusfaustino** [comment]
Thank you for you reply, I will try to bypass it and see if it compiles or try to implement it on the lib. ^^

**Comment by matheusfaustino** [comment]
Well, I already modified the lib to handle `#pragma once`. I'm studying the case from #8 and after that I will submit a PR for this issue.

**Comment by bwoebi** [comment]
pragma once has been implemented as part of #17, the other pragmas are skipped for now. Closing.
